### PR TITLE
Remove runconfig.Merge

### PR DESCRIPTION
**Pull request by anusha-ragunathan**
Merge was used by builder and daemon. With this commit, the builder
call has been inlined and the function moved to the daemon package,
which is the only other caller.

Signed-off-by: Anusha Ragunathan anusha@docker.com

### CODE REVIEW

**Comment by vdemeester** [comment]
LGTM 🐼

**Comment by runcom** [comment]
This https://github.com/docker/docker/blob/master/runconfig/merge_test.go has to be (re)moved as well

**Comment by vdemeester** [comment]
@runcom oh.. you have the eye :stuck_out_tongue:

File: builder/dockerfile/dispatchers.go [hunk 1]
```diff
@@ -317,7 +316,10 @@ func run(b *Builder, args []string, attributes map[string]bool, original string)
 
 	// stash the cmd
 	cmd := b.runConfig.Cmd
-	runconfig.Merge(b.runConfig, config)
+	if b.runConfig.Entrypoint.Len() == 0 && b.runConfig.Cmd.Len() == 0 {
+		b.runConfig.Cmd = config.Cmd
```
Comment (MHBauer) [inline builder/dockerfile/dispatchers.go]: So the only thing being merged here is the `Cmd`? It looks like that to me, I want to confirm.

Comment (anusha-ragunathan) [inline builder/dockerfile/dispatchers.go]: Yes. config is created (L312) with a Cmd and Image, nothing more. Merge() doesnt mutate Image, only Cmd. So we are only merging Cmd here.

**Comment by MHBauer** [comment]
I like it so far. Handle the test and I'll take another look after.

**Comment by calavera** [comment]
Unit tests are failing:

```
# github.com/docker/docker/runconfig
runconfig/merge_test.go:34: undefined: Merge
runconfig/merge_test.go:72: undefined: Merge
```

**Comment by anusha-ragunathan** [comment]
@calavera : Yes, @runcom mentioned it. I'm on it.

File: daemon/commit.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 	"time"
 
 	"github.com/docker/docker/api/types"
+	contapi "github.com/docker/docker/api/types/container"
```
Comment (tiborvass) [inline daemon/commit.go]: `containertypes` would be better (as done in other files)

**Comment by runcom** [comment]
LGTM now if janky's happy :)

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**